onnxruntime/core/providers/coreml/coreml_execution_provider.h

```diff
@@ -12,9 +12,14 @@ namespace coreml {
 class Model;
 }
 
+struct CoreMLOptions {
+  uint32_t coreml_flags = 0;
+  std::string cache_path;
```

Comment:
nit: can we keep changes to add supporting a cache path in a separate PR? 

Also wondering if this is going to grow better if we keep it separate from coreml_flags. It's a little hard to see what is supported with the current setup. If we define the full set of options in this struct, and we populate those from either the flags, or the provider options, it's clear what the full set of options is.

I'd suggest making CoreMLOptions a class and make it responsible for validation. Can have a ctor for flags and one for provider options. That way all the validation is done in one place and it's clear which class is responsible for validation.

Reply:
Yeah, it's good to seperate  cache path in another PR.

If we make different flag has its own variable to mark it.  How can we capatiable it with current `coreml_flags`?
Besides, we can easily pass coreml_flags(uint32_t) to object-c.  can we pass a class from c++ to obect-c directly?